src/lib/operators/abstract_operator.cpp

```diff
@@ -297,4 +308,24 @@ std::ostream& operator<<(std::ostream& stream, const AbstractOperator& abstract_
   return stream;
 }
 
+void AbstractOperator::_transition_to(OperatorState new_state) {
+  OperatorState previous_state = _state.exchange(new_state);
+
+  // Check validity of state transition
+  switch (new_state) {
+    case OperatorState::Running:
+      Assert(previous_state == OperatorState::Created, "Illegal state transition to OperatorState::Running");
+      break;
+    case OperatorState::Executed:
+      Assert(previous_state == OperatorState::Running, "Illegal state transition to OperatorState::Executed");
+      break;
+    case OperatorState::Cleared:
+      Assert(previous_state == OperatorState::Executed || previous_state == OperatorState::Cleared,
```

Comment:
Why would it be legal to clear an operator twice?

Reply:
Answered above